conda_lock/conda_lock.py

```diff
@@ -23,42 +31,130 @@
 DEFAULT_PLATFORMS = ["osx-64", "linux-64", "win-64"]
 
 
-def solve_specs_for_arch(channels: List[str], specs: List[str], platform: str) -> str:
+def ensure_conda(conda_executable: Optional[str]):
+    if conda_executable:
+        if pathlib.Path(conda_executable).exists():
+            return conda_executable
+
+    conda_executable = shutil.which("conda")
+    if conda_executable:
+        return conda_executable
+
+    conda_executable = shutil.which("conda.exe")
+    if conda_executable:
+        return conda_executable
+
+    logging.info(
+        "No existing conda installation found.  Installing the standalone conda solver"
+    )
+    return install_conda_exe()
+
+
+def install_conda_exe():
+    conda_exe_prefix = "https://repo.anaconda.com/pkgs/misc/conda-execs"
+    if platform.system() == "Linux":
+        conda_exe_file = "conda-latest-linux-64.exe"
+    elif platform.system() == "Darwin":
+        conda_exe_file = "conda-latest-osx-64.exe"
+    elif platform.system() == "NT":
+        conda_exe_file = "conda-latest-win-64.exe"
+    else:
+        # TODO: Support windows here
+        raise ValueError(f"Unsupported platform: {platform.system()}")
+
+    resp = requests.get(f"{conda_exe_prefix}/{conda_exe_file}", allow_redirects=True)
+    resp.raise_for_status()
+    target_filename = os.path.expanduser(pathlib.Path(__file__).parent / "conda.exe")
+    with open(target_filename, "wb") as fo:
+        fo.write(resp.content)
+    st = os.stat(target_filename)
+    os.chmod(target_filename, st.st_mode | stat.S_IXUSR)
+    return target_filename
+
+
+CONDA_PKGS_DIRS = None
+
+
+def conda_pkgs_dir():
```

Comment:
Does it make sense to upstream that to `conda` itself? Smells like a workaround to something that should be in conda.

Reply:
There is an open issue on conda about this already https://github.com/conda/conda/issues/6902

Reply:
2018!